ProcessMaker/processmaker
# Fix request lock logic, add multiinstance

## Issue & Reproduction Steps
When multiple instances (scripts, service tasks, call activities, etc) are running in parallel the lock logic for one or multiple request running in a collaborations were failing:

- Lock only blocks one request at a time, but for a collaborations all the requests should be locked.
- Bug in lock time out query
- When a Job takes many time to get the Lock, the request could changed, it should be reloaded

## Solution
- Implement a Lock for all the requests in a collaboration.
- Fix query to get due locks.
- Reload data and tokens when a Lock is obtained.

## How to Test
Run some of the following processes:

![image](https://user-images.githubusercontent.com/8028650/143130524-d33d6456-01d4-4592-9716-8b28ff15bc65.png)

1. Run the first process, 10 script tasks should be executed, then the process Completed.
2. Run the second process, 10 sub processes should be executed, the Signal is triggered, then all the MultiInstance Sub Processes are closed, and continue to the task "**Catch Task**"

[MultiInstance-processes.json.txt](https://github.com/ProcessMaker/processmaker/files/7591069/MultiInstance-processes.json.txt)

## Related Tickets & Packages
- [FOUR-4407](https://processmaker.atlassian.net/browse/FOUR-4407)
- [FOUR-4597](https://processmaker.atlassian.net/browse/FOUR-4597)

## Code Review Checklist
